crates/database/src/jit_orders.rs

```diff
@@ -0,0 +1,258 @@
+    pub kind: OrderKind,
+    pub partially_fillable: bool,
+    pub signature: Vec<u8>,
+    pub sum_sell: BigDecimal,
+    pub sum_buy: BigDecimal,
+    pub sum_fee: BigDecimal,
+    pub receiver: Address,
+    pub signing_scheme: SigningScheme,
+    pub sell_token_balance: SellTokenSource,
+    pub buy_token_balance: BuyTokenDestination,
+    pub executed_surplus_fee: BigDecimal,
+}
+
+/// 1:1 mapping to the `jit_orders` table, used to store orders.
+#[derive(Debug, Clone, Default, PartialEq, sqlx::FromRow)]
+pub struct JitOrder {
+    pub block_number: i64,
+    pub log_index: i64,
+    pub uid: OrderUid,
+    pub owner: Address,
+    pub creation_timestamp: DateTime<Utc>,
+    pub sell_token: Address,
+    pub buy_token: Address,
+    pub sell_amount: BigDecimal,
+    pub buy_amount: BigDecimal,
+    pub valid_to: i64,
+    pub app_data: AppId,
+    pub fee_amount: BigDecimal,
+    pub kind: OrderKind,
+    pub partially_fillable: bool,
+    pub signature: Vec<u8>,
+    pub receiver: Address,
+    pub signing_scheme: SigningScheme,
+    pub sell_token_balance: SellTokenSource,
+    pub buy_token_balance: BuyTokenDestination,
+}
+
+pub async fn upsert_orders(
+    ex: &mut PgConnection,
+    jit_orders: &[JitOrder],
+) -> Result<(), sqlx::Error> {
+    if jit_orders.is_empty() {
+        return Ok(());
+    }
+
+    let mut query_builder = QueryBuilder::new(
+        r#"
+        INSERT INTO jit_orders (
+            block_number,
+            log_index,
+            uid,
+            owner,
+            creation_timestamp,
+            sell_token,
+            buy_token,
+            sell_amount,
+            buy_amount,
+            valid_to,
+            app_data,
+            fee_amount,
+            kind,
+            partially_fillable,
+            signature,
+            receiver,
+            signing_scheme,
+            sell_token_balance,
+            buy_token_balance
+        ) 
+        "#,
+    );
+
+    query_builder.push_values(jit_orders.iter(), |mut builder, jit_order| {
+        builder
+            .push_bind(jit_order.block_number)
+            .push_bind(jit_order.log_index)
+            .push_bind(jit_order.uid)
+            .push_bind(jit_order.owner)
+            .push_bind(jit_order.creation_timestamp)
+            .push_bind(jit_order.sell_token)
+            .push_bind(jit_order.buy_token)
+            .push_bind(jit_order.sell_amount.clone())
+            .push_bind(jit_order.buy_amount.clone())
+            .push_bind(jit_order.valid_to)
+            .push_bind(jit_order.app_data)
+            .push_bind(jit_order.fee_amount.clone())
+            .push_bind(jit_order.kind)
+            .push_bind(jit_order.partially_fillable)
+            .push_bind(jit_order.signature.clone())
+            .push_bind(jit_order.receiver)
+            .push_bind(jit_order.signing_scheme)
+            .push_bind(jit_order.sell_token_balance)
+            .push_bind(jit_order.buy_token_balance);
+    });
+
+    query_builder.push(
```

Comment:
not following? Ilya previously suggested to have a function that inserts batch of jit orders.

Reply:
I mean why do we need to manage conflicts here?

Reply:
I think this is a leftover from when the OnSettlementEventUpdater was doing reindexing and in case of some bug in event indexing logic we might get stuck on a same insert.
I am also fine putting `ON CONFLICT DO NOTHING` if it seems cleaner in this case?